mismi-s3-core/src/Mismi/S3/Core/Data.hs

```diff
@@ -212,6 +229,19 @@ addressFromText :: Text -> Maybe Address
 addressFromText =
   rightToMaybe . parseOnly s3Parser
 
+locationToText :: Location -> Text
+locationToText loc =
+  case loc of
+    LocalLocation fp -> T.pack fp
+    S3Location a -> addressToText a
+
+-- | Parse a location. If it parses as a valid s3 address, it gets parsed as an
+-- 'S3Location' otherwise it is assumed to be a 'FilePath' on the local
+-- filesystem.
```

Comment:
Is there not some sensible parsing / validation that could be done if it is a file path?

Reply:
It potentially could check for illegal file name characters or something, but there is still no easy way to report error. If we accept anything here, then error can be reported quite easily when the `FilePath` is used.

Reply:
I didn't think there was any illegal characters for a posix path, it's just bytes

Reply:
I think @jystic is right. Ifs its a posix path then anything goes.

Reply:
@jystic almost, `\0` is not allowed. POSIX paths are pretty savage.

Reply:
yolo paths it is then. 👍